src/ch/imedias/rsccfx/model/util/KeyUtil.java

```diff
@@ -0,0 +1,128 @@
+package ch.imedias.rsccfx.model.util;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.Streams;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import javafx.beans.binding.Bindings;
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.ReadOnlyBooleanProperty;
+import javafx.beans.property.ReadOnlyStringProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.property.StringProperty;
+
+/**
+ * Handles everything related to the key.
+ */
+public class KeyUtil {
+
+  private static final Logger LOGGER =
+      Logger.getLogger(KeyUtil.class.getName());
+
+  private static final String KEY_FORMAT_DELIMITER = " ";
+  private static final int KEY_FORMAT_DELIMITER_EVERY = 3;
+
+  private final StringProperty key = new SimpleStringProperty();
+  private final StringProperty formattedKey = new SimpleStringProperty();
+  private final BooleanProperty keyValid = new SimpleBooleanProperty(false);
+
+  public KeyUtil() {
+    setupBindings();
+  }
+
+  private void setupBindings() {
+    // if the key gets changed, this sets the formatted key
+    formattedKey.bind(
+        Bindings.createStringBinding(
+            () -> formatKey(getKey()), key
+        )
+    );
+
+    // if the key gets changed, the key will get validated
+    keyValid.bind(
+        Bindings.createBooleanBinding(
+            () -> validateKey(getKey()), key
+        )
+    );
+  }
+
+  /**
+   * Formats the key, so it has spaces every 3 characters.
+   *
+   * @param key the unformatted key which should be formatted.
+   * @return the formatted key, with spaces every 3 characters.
+   */
+  public String formatKey(String key) {
+    if (key != null) {
+      key = deformatKey(key); // make sure the key doesn't have spaces in it already
+      // shorten key to 9 characters if it's too long
+      if (key.length() > 9) {
+        key = key.substring(0, 9);
```

Comment:
What if we add something at the beginning? Does simply the last number get cut off?